### ISSUE TITLE
Find state of persistence startup status

### ISSUE THREAD

**Issue by Marmoset-threat** [openhab/openhab-core#3244]
**The Problem:**

On slower systems with many items, it is easily possible that the rules load and the rules engine begins operating _before_ items are restored to their previous state via a persistence service during OH startup procedures. These rules then execute on null-state items, and this has the potential to cause all sorts of issues

**Potential solutions:**

- Implement a global setting "Block rule triggers until persistence services fully loaded"
- Allow the user to check the current state of persistence services in rules, similar to `getThingStatusInfo` but maybe `getPersistenceStatusInfo(mapdb)`

**Advantages:**

I think being able to interact with persistence services in such a way and specifically wait for OH to be ready after a startup would be a solid step towards making the OH system more robust and it would simplify debugging issues with persistence services.

**Comment by J-N-K** [comment]
I don't think that we should introduce such methods. But maybe we can add a `SystemStartLevelCondition` so you can prevent execution prior to a given start level.

**Comment by mhilbush** [comment]
Wasn't start level 30 (START_LEVEL_STATES) intended to ensure that item states have been restored from the persistence service(s)?

**Comment by J-N-K** [comment]
Yes, but I would prefer to have a more generic version (i.e. a configurable start-level). This would also solve #3073.

**Comment by presidentio** [comment]
I also faced this issue after upgrading to Openhab 4. For me, it triggers a lot of rules that were created to react to the change of an item state. I'd like to have it by default or to have a global config that makes sure all items are restored by the persistence service before rules are initialized.

**Comment by rkoshak** [comment]
In the mean time, see https://community.openhab.org/t/delay-start-4-0-0-0-4-1-0-0/144884 as a work around.
